File: identity/go/kernaq/liveness.go
package kernaq

import (
	"context"
	"fmt"
)

// LivenessResource handles stateless liveness detection.
type LivenessResource struct {
	c *client
}

// Check checks for a live real person. Supply Video OR all three Frame fields.
// Nothing stored.
func (r *LivenessResource) Check(ctx context.Context, req LivenessCheckRequest) (*LivenessCheckResponse, error) {
	var files []filePart
	if req.Video != nil {
		vidName := nameOr(req.VideoName, "liveness.mp4")
		files = []filePart{{field: "video", reader: req.Video, filename: vidName, mime: mimeFromName(vidName)}}
	} else if req.Frame1 != nil && req.Frame2 != nil && req.Frame3 != nil {
		f1 := nameOr(req.Frame1Name, "frame1.jpg")
		f2 := nameOr(req.Frame2Name, "frame2.jpg")
		f3 := nameOr(req.Frame3Name, "frame3.jpg")
		files = []filePart{
			{field: "frame1", reader: req.Frame1, filename: f1, mime: "image/jpeg"},
			{field: "frame2", reader: req.Frame2, filename: f2, mime: "image/jpeg"},
			{field: "frame3", reader: req.Frame3, filename: f3, mime: "image/jpeg"},
		}
	} else {
		return nil, fmt.Errorf("supply either Video or all three Frame fields (Frame1, Frame2, Frame3)")
	}
	var out LivenessCheckResponse
	if err := r.c.upload(ctx, "/liveness/check", nil, files, &out); err != nil {
		return nil, err
	}
	return &out, nil
}

File: identity/node/src/liveness.ts
import { BaseClient, type FileInput } from './client.js'
import type { LivenessCheckRequest, LivenessCheckResponse } from './types.js'

export class LivenessResource extends BaseClient {

  /**
   * Check a video (or frame sequence) for liveness.
   * Detects replay attacks and spoof attempts (synchronous).
   *
   * Supply `video` OR `frame1` + `frame2` + `frame3`.
   *
   * @example
   * const result = await kernaq.liveness.check({
   *   video: fs.createReadStream('liveness.mp4'),
   * })
   * console.log(result.passed) // true
   */
  check(req: LivenessCheckRequest): Promise<LivenessCheckResponse> {
    const files: Array<{ field: string; value: FileInput; filename: string; contentType: string }> = []

    if (req.video) {
      files.push({
        field:       'video',
        value:       req.video as FileInput,
        filename:    req.videoName ?? 'liveness.mp4',
        contentType: 'video/mp4',
      })
    } else if (req.frame1 && req.frame2 && req.frame3) {
      files.push(
        { field: 'frame_1', value: req.frame1 as FileInput, filename: req.frame1Name ?? 'frame_1.jpg', contentType: 'image/jpeg' },
        { field: 'frame_2', value: req.frame2 as FileInput, filename: req.frame2Name ?? 'frame_2.jpg', contentType: 'image/jpeg' },
        { field: 'frame_3', value: req.frame3 as FileInput, filename: req.frame3Name ?? 'frame_3.jpg', contentType: 'image/jpeg' },
      )
    }

    return this.upload<LivenessCheckResponse>('/liveness/check', {}, files)
  }
}
